scheduling-manager/src/main/java/com/amazonaws/blox/scheduling/reconciler/CloudWatchEvent.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.scheduling.reconciler;
+
+import com.amazonaws.regions.Regions;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.time.Instant;
+import java.util.List;
+import lombok.Data;
+
+/**
+ * Representation of an Event from Cloudwatch.
+ *
+ * <p>See http://docs.aws.amazon.com/AmazonCloudWatch/latest/events/EventTypes.html for details.
+ *
+ * @param <T> The type of the "detail" field.
+ */
+@Data
+public class CloudWatchEvent<T> {
```

Comment:
why not use the provided type so it's kept up to date? (https://github.com/aws/aws-lambda-java-libs/blob/master/aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ScheduledEvent.java)

Reply:
Oh! I couldn't find that when I went looking, will update.

Reply:
Ah, it looks like that is in the 2.0.0 version of the events package, which was released after I started working on this. Will have to upgrade to that version first.

Reply:
Aaaand that one doesn't play nicely with the Jackson serializer, as caught by ReconcilerEntryPointTest:

```
com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "version" (class com.amazonaws.services.lambda.runtime.events.ScheduledEvent), not marked as ignorable (8 known properties: "detail", "region", "resources", "detailType", "account", "source", "time", "id"])
 at [Source: java.io.BufferedInputStream@1cb96; line: 2, column: 15] (through reference chain: com.amazonaws.services.lambda.runtime.events.ScheduledEvent["version"])
```

I'll dig into it a bit, I wonder if that works in Lambda.

Reply:
Looks like they have some internal Jackson magic to make it work. I think I'm just going to rip out the JacksonRequestStreamHandler for this function, and use the POJO request handler instead.

Reply:
This is proving more difficult than I thought. I've broken out a task for it in the issue to come back after this PR.